### ISSUE TITLE
Dependency resolved to the lowest version

### ISSUE THREAD

**Issue by bzmario** [CocoaPods/CocoaPods#8510]
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

Run `pod install`

## What did you expect to happen?

All dependencies are installed the latest version, even their versions are not specified.

## What happened instead?

All private dependency pods are resolved to the latest version, except only one dependency always resolved to the lowest version.

### Stack

```
   CocoaPods : 1.6.0
        Ruby : ruby 2.5.3p105 (2018-10-18 revision 65156) [x86_64-darwin17]
    RubyGems : 2.7.6
        Host : Mac OS X 10.13.6 (17G5019)
       Xcode : 10.1 (10B61)
         Git : git version 2.17.2 (Apple Git-113)
Ruby lib dir : /Users/mario/.rbenv/versions/2.5.3/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 682191c942dee6f2b819248c700e74a99556ca24
               qima-inc-applib-cocoapodsspecs - http://gitlab.abc.com/CocoaPodsSpecs.git @ b204d3ac92174b6483b4cd4918e6d880ebea73f8
               qima-inc-applib-renrenpodspecs - http://gitlab.abc.com/PrivatePodspecs.git @ b591d5e1f3ff04ffcc609a6bd8b2d02ae9bb85ed
```

### Installation Source

```
Executable Path: /Users/abcd/.rbenv/versions/2.5.3/bin/pod
```

### Plugins

```
cocoapods-deintegrate  : 1.0.2
cocoapods-dependencies : 1.3.0
cocoapods-plugins      : 1.0.0
cocoapods-search       : 1.0.0
cocoapods-stats        : 1.0.0
cocoapods-trunk        : 1.3.1
cocoapods-try          : 1.1.0
```

### Podfile

```ruby
source 'http://gitlab.abc.com/PrivatePodspecs.git'
source 'http://gitlab.abc.com/CocoaPodsSpecs.git'

target 'LoggerDemo' do
  # Uncomment the next line if you're using Swift or would like to use dynamic frameworks
  # use_frameworks!

  # Pods for LoggerDemo
  pod 'Logger', '~> 0.2.10'

end
```

Private pod Logger's podsepc

```
Pod::Spec.new do |s|
  s.name             = 'Logger'
  s.version          = '0.2.10'
  s.homepage         = 'http://gitlab.qima-inc.com/Logger'
  s.license          = { :type => 'MIT', :file => 'LICENSE' }
  s.source           = { :git => 'git@gitlab.abc.com:abc/Logger.git', :tag => s.version.to_s }

  s.ios.deployment_target = '8.0'

  s.source_files = 'Logger/Logger/Classes/**/*'
  
  s.dependency  'ServiceA'
  s.dependency  'ServiceA'
  s.dependency  'Uploader'
end
```

Service  A and B are resolved to the latest version, and Uploader is the lowest version.

**Comment by dnkoutso** [comment]
This isn't enough information to work with. What versions are published what is the deployment target of your app?

Please upload a sample app demonstrating the issue if possible.

**Comment by bzmario** [comment]
The demo is just an empty Xcode project. The source and lib are private, so a sample project won't work in public.

Deployment target is `LoggerDemo`, which depends on `Logger`.  And `Logger` depends on `ServiceA`, `Service B` and `Uploader`.

`Service A` and `Service B` are resolved to the latest version. 

`Uploader`'s latest version is `1.7.0`, resolved to `0.1.0`.  I can specify its version with `~> 1.7.0`, so it can be resolved correctly.

Just don't know why cocoapods get the lowest version just for `Uploader`.

**Comment by segiddins** [comment]
We will need a sample app (you can create a demo that just reproduces this issue if you can't share your original project) to really be able to investigate, sorry
